### ISSUE TITLE
Problems with training SST-2

### ISSUE THREAD

**Issue by gaopengcuhk** [facebookresearch/XLM#152]
Hi, I test the pretrained BERT finetuning SST-2 task. However, I find the validation accuracy is the same for each epoch. Can you check the reason?

Command:
srun --gres gpu:1 -p clusterRTX python glue-xnli.py --exp_name test_xlm_en_glue  --dump_path ./dumped/                                                                                                                                                                 --model mlm_en_2048.pth --data_path ./data/processed/XLM_en/ --transfer_tasks SST-2,QNLI --optimizer_p adam,lr=0.000125 --fine                                                                                                                                                                tune_layers "0:_1"  --batch_size 8 --n_epochs 250 --epoch_size 20000 --max_len 256 --max_vocab -1

Output:
NFO - 08/01/19 11:33:47 - 1:07:01 - GLUE - SST-2 - Epoch 6 - Train iter   19926 - 421.6 words/s - XE Loss: 0.6889
INFO - 08/01/19 11:33:50 - 1:07:03 - GLUE - SST-2 - Epoch 6 - Train iter   20006 - 481.3 words/s - XE Loss: 0.7019
INFO - 08/01/19 11:33:50 - 1:07:03 - GLUE - SST-2 - Evaluating epoch 6 ...
INFO - 08/01/19 11:33:56 - 1:07:09 - __log__:{"epoch": 6, "valid_sst-2_acc": 50.91743119266055, "valid_sst-2_f1": 67.477203647                                                                                                                                                                4164, "valid_sst-2_mc": 0.0}
INFO - 08/01/19 11:33:56 - 1:07:09 - Wrote 872 valid predictions to ./dumped/test_xlm_en_glue/115719/valid.pred.6
INFO - 08/01/19 11:34:09 - 1:07:22 - Wrote 1821 test predictions to ./dumped/test_xlm_en_glue/115719/test.pred.6

Explanination
The valid is the same for every epoch.

I download the SST-2 dataset using the provided bash file. However, only train, dev and test file is generated. I rename the dev file to valid file.

**Comment by gaopengcuhk** [comment]
Hi

I meet the same problem for the rest of GLUE tasks. It seems the finetuning failed in GLUE. Can you look into the problem and I will provide any information you need.

**Comment by gaopengcuhk** [comment]
I can solve the problem by fine-tuning partial layers instead of all layers. Can you share your parameters for getting the GLUE score?

**Comment by aconneau** [comment]
@gaopengcuhk : could you try different learning rates (0.000005, 0.000025) and check if your data has been preprocessed correctly?

**Comment by aconneau** [comment]
Could you try to use these parameters:

    "optimizer_e": ["adam,lr=0.00000125", "adam,lr=0.0000025", "adam,lr=0.000005"]
    "optimizer_p": ["adam,lr=0.000005", "adam,lr=0.000025", "adam,lr=0.000125"]

It seems that the README doesn't include the parameter "optimizer_e" which is a mistake. Thank you for flagging this issue. Could you try with the parameters above and let me know if that works?

**Comment by gaopengcuhk** [comment]
Thank you so much for your reply. Finetuning works right now. I will report my benchmark here in a few days. 

By the way, lr of optimizer_e must be smaller than optimizer_p for the model to converge.

**Comment by gaopengcuhk** [comment]
Can you also look into the following issue?
https://github.com/facebookresearch/XLM/issues/159

It seems layer deeper than 8 will diverge in the pretraining stage. By diverge, the loss will go around 7 never go down.

**Comment by gaopengcuhk** [comment]
SST-2 dataset will get the following performance for only one epoch

__log__:{"epoch": 0, "valid_sst-2_acc": 94.26605504587155, "valid_sst-2_f1": 94.34389140271495, "valid_sst-2_mc": 88.5339065020795}

**Comment by aconneau** [comment]
great! thank you for flagging this issue and sry for the oversight in the README.

**Comment by gaopengcuhk** [comment]
After training several epoches on SST-2, the loss will explosion then go to normal. The final performance is around 94 without too much parameter fine-tuning. Is this normal?


INFO - 08/05/19 13:11:36 - 0:57:06 - GLUE - SST-2 - Epoch 5 - Train iter   18000 - 493.1 words/s - XE Loss: 0.1657
INFO - 08/05/19 13:11:38 - 0:57:08 - GLUE - SST-2 - Epoch 5 - Train iter   18080 - 413.7 words/s - XE Loss: 0.1317
INFO - 08/05/19 13:11:40 - 0:57:10 - GLUE - SST-2 - Epoch 5 - Train iter   18160 - 481.2 words/s - XE Loss: 0.1740
INFO - 08/05/19 13:11:43 - 0:57:13 - GLUE - SST-2 - Epoch 5 - Train iter   18240 - 484.8 words/s - XE Loss: 0.2652
INFO - 08/05/19 13:11:45 - 0:57:15 - GLUE - SST-2 - Epoch 5 - Train iter   18320 - 417.7 words/s - XE Loss: 0.3179
INFO - 08/05/19 13:11:47 - 0:57:17 - GLUE - SST-2 - Epoch 5 - Train iter   18400 - 473.9 words/s - XE Loss: 0.4944
INFO - 08/05/19 13:11:50 - 0:57:20 - GLUE - SST-2 - Epoch 5 - Train iter   18480 - 460.6 words/s - XE Loss: 0.7642
INFO - 08/05/19 13:11:52 - 0:57:22 - GLUE - SST-2 - Epoch 5 - Train iter   18560 - 444.8 words/s - XE Loss: 0.8530
INFO - 08/05/19 13:11:54 - 0:57:24 - GLUE - SST-2 - Epoch 5 - Train iter   18640 - 472.2 words/s - XE Loss: 0.7591
INFO - 08/05/19 13:11:57 - 0:57:27 - GLUE - SST-2 - Epoch 5 - Train iter   18720 - 479.6 words/s - XE Loss: 0.7232
INFO - 08/05/19 13:11:59 - 0:57:29 - GLUE - SST-2 - Epoch 5 - Train iter   18800 - 532.3 words/s - XE Loss: 0.7052
INFO - 08/05/19 13:12:01 - 0:57:31 - GLUE - SST-2 - Epoch 5 - Train iter   18880 - 412.5 words/s - XE Loss: 0.7330
INFO - 08/05/19 13:12:03 - 0:57:33 - GLUE - SST-2 - Epoch 5 - Train iter   18960 - 433.6 words/s - XE Loss: 0.6872
INFO - 08/05/19 13:12:05 - 0:57:35 - GLUE - SST-2 - Epoch 5 - Train iter   19040 - 461.8 words/s - XE Loss: 0.6760
INFO - 08/05/19 13:12:07 - 0:57:38 - GLUE - SST-2 - Epoch 5 - Train iter   19120 - 475.0 words/s - XE Loss: 0.8064
INFO - 08/05/19 13:12:10 - 0:57:40 - GLUE - SST-2 - Epoch 5 - Train iter   19200 - 442.6 words/s - XE Loss: 0.8201
INFO - 08/05/19 13:12:12 - 0:57:42 - GLUE - SST-2 - Epoch 5 - Train iter   19280 - 444.2 words/s - XE Loss: 0.7902
INFO - 08/05/19 13:12:14 - 0:57:45 - GLUE - SST-2 - Epoch 5 - Train iter   19360 - 487.2 words/s - XE Loss: 0.6828
INFO - 08/05/19 13:12:16 - 0:57:47 - GLUE - SST-2 - Epoch 5 - Train iter   19440 - 437.0 words/s - XE Loss: 0.7175
INFO - 08/05/19 13:12:19 - 0:57:49 - GLUE - SST-2 - Epoch 5 - Train iter   19520 - 436.2 words/s - XE Loss: 0.6916
INFO - 08/05/19 13:12:21 - 0:57:51 - GLUE - SST-2 - Epoch 5 - Train iter   19600 - 501.7 words/s - XE Loss: 0.7160
INFO - 08/05/19 13:12:23 - 0:57:53 - GLUE - SST-2 - Epoch 5 - Train iter   19680 - 418.9 words/s - XE Loss: 0.6931
INFO - 08/05/19 13:12:25 - 0:57:56 - GLUE - SST-2 - Epoch 5 - Train iter   19760 - 502.3 words/s - XE Loss: 0.6825
INFO - 08/05/19 13:12:28 - 0:57:58 - GLUE - SST-2 - Epoch 5 - Train iter   19840 - 491.5 words/s - XE Loss: 0.6993
INFO - 08/05/19 13:12:30 - 0:58:00 - GLUE - SST-2 - Epoch 5 - Train iter   19920 - 460.7 words/s - XE Loss: 0.7098
INFO - 08/05/19 13:12:32 - 0:58:02 - GLUE - SST-2 - Epoch 5 - Train iter   20000 - 490.8 words/s - XE Loss: 0.7855
INFO - 08/05/19 13:12:32 - 0:58:02 - GLUE - SST-2 - Evaluating epoch 5 ...
/homes/pgao/.conda/envs/py36/lib/python3.6/site-packages/sklearn/metrics/classification.py:872: RuntimeWarning: invalid value encountered in double_scalars
  mcc = cov_ytyp / np.sqrt(cov_ytyt * cov_ypyp)
INFO - 08/05/19 13:12:38 - 0:58:09 - __log__:{"epoch": 5, "valid_sst-2_acc": 50.91743119266055, "valid_sst-2_f1": 67.4772036474164, "valid_sst-2_mc": 0.0}

**Comment by aconneau** [comment]
No this is not normal, this may be linked to the learning rate. In general, consider sweeping on the learning rate parameter for fine-tuning as is specified here: https://github.com/facebookresearch/XLM#3-fine-tune-a-pretrained-model-on-glue-tasks.

**Comment by cocaer** [comment]
@aconneau The download link in get-data-glue.sh  is expired.  Would  you like to update the link?
